### PR TITLE
Initial workload module

### PR DESCRIPTION
**Pull request by vaporos**

### CODE REVIEW

File: libtransact/src/workload/error.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+use std;
+use std::error::Error as StdError;
+
+use crate::batch::BatchBuildError;
+use crate::transaction::TransactionBuildError;
+
+#[derive(Debug)]
+pub enum WorkloadError {
+    // Returned when an error occurs when using BatchBuilder.
+    BatchBuildError(Box<BatchBuildError>),
```
Comment (peterschwarz) [inline libtransact/src/workload/error.rs]: I don't think these need to be Boxed

File: libtransact/src/workload/error.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+impl std::error::Error for WorkloadError {
+    fn description(&self) -> &str {
+        match *self {
+            WorkloadError::BatchBuildError(ref err) => err.description(),
+            WorkloadError::TransactionBuildError(ref err) => err.description(),
+        }
+    }
+
+    fn cause(&self) -> Option<&std::error::Error> {
```
Comment (peterschwarz) [inline libtransact/src/workload/error.rs]: Consider implementing this via `source` (see https://doc.rust-lang.org/std/error/trait.Error.html#method.source), and dropping the above description, as it's "Soft-deprecated"

Comment (vaporos) [reply libtransact/src/workload/error.rs]: I assume you mean dropping cause, not description?

File: libtransact/src/workload/error.rs (hunk 1 above)
Comment (vaporos) [reply libtransact/src/workload/error.rs]: Correct, this is no longer needed. Fixed.

File: libtransact/src/workload/error.rs (hunk 2 above)
Comment (vaporos) [reply libtransact/src/workload/error.rs]: Replace cause() with source().

File: libtransact/src/workload/xo.rs [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+impl TransactionWorkload for XoTransactionWorkload {
+    fn next_transaction(&mut self) -> Result<TransactionPair, WorkloadError> {
+        let nonce = self
+            .rng
+            .sample_iter(&Alphanumeric)
+            .take(NONCE_SIZE)
+            .collect::<String>()
+            .as_bytes()
```
Comment (peterschwarz) [inline libtransact/src/workload/xo.rs]: These two calls can be replaced with `into_bytes()`

File: libtransact/src/workload/xo.rs [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    CREATE,
+}
+
+struct Payload {
+    name: String,
+    action: Action,
+}
+
+impl Payload {
+    pub fn new_as_create_with_random_name(rnd: &mut Hc128Rng) -> Self {
```
Comment (peterschwarz) [inline libtransact/src/workload/xo.rs]: Can you add some doc comments to these `new` functions? It is not entirely obvious what `as_create` means.

Comment (vaporos) [reply libtransact/src/workload/xo.rs]: Added

File: libtransact/src/workload/xo.rs (hunk 3 above)
Comment (vaporos) [reply libtransact/src/workload/xo.rs]: Fixed

**Pull request merged**